test/test.js

```diff
@@ -1568,13 +1570,75 @@ test('simulate invalid alter _lightMyRequest.isDone without end', (t) => {
 })
 
 test('no error for response destory', (t) => {
+  t.plan(1)
+
   const dispatch = function (req, res) {
     res.destroy()
   }
 
   inject(dispatch, { method: 'GET', url: '/' }, (err, res) => {
     t.error(err)
   })
+})
 
-  t.end()
+test('request destory without error', (t) => {
+  t.plan(2)
+
+  const dispatch = function (req, res) {
+    req.destroy()
+  }
+
+  inject(dispatch, { method: 'GET', url: '/' }, (err, res) => {
+    t.error(err)
+    t.equal(res, null)
+  })
+})
+
+test('request destory with error', (t) => {
+  t.plan(2)
+
+  const fakeError = new Error('some-err')
+
+  const dispatch = function (req, res) {
+    req.destroy(fakeError)
+  }
+
+  inject(dispatch, { method: 'GET', url: '/' }, (err, res) => {
+    t.equal(err, fakeError)
+    t.equal(res, null)
+  })
+})
+
+test('compatible with stream.finished', (t) => {
+  t.plan(3)
+
+  const dispatch = function (req, res) {
+    finished(res, (err) => {
+      t.ok(err instanceof Error)
+    })
+
+    req.destroy()
+  }
+
+  inject(dispatch, { method: 'GET', url: '/' }, (err, res) => {
+    t.error(err)
+    t.equal(res, null)
+  })
+})
+
+test('compatible with eos', (t) => {
+  t.plan(3)
+
+  const dispatch = function (req, res) {
+    eos(res, (err) => {
+      t.ok(err instanceof Error)
```

Comment:
eos thinks this `destroyed` request is indeed premature close. However, we're not interested in what the error is, but the fact that this callback is called. Right?

Reply:
It is correct that the error is 'premature close' as there is no error. Add another test that checks that the error is passed through correctly